ruby-challenges.rb

```diff
@@ -5,28 +5,73 @@
 
 beverages_array = ['coffee', 'tea', 'juice', 'water', 'soda water']
 
-letter_o = 'o'
-# Expected output: ['coffee', 'soda water']
-letter_t = 't'
-# Expected output: ['tea', 'water', 'soda water']
+# letter_o = 'o'
+# # Expected output: ['coffee', 'soda water']
+# letter_t = 't'
+# # Expected output: ['tea', 'water', 'soda water']
+
+
+def word_with_letters(givenArray, string)
+  givenArray.filter {|newArray| newArray.include? string}
+end
+p word_with_letters(beverages_array, letter_o)
+p word_with_letters(beverages_array, letter_t)
+#output= ["coffee", "soda water"]
+#output= ["tea", "water", "soda water"]
+
 
 
 # -------------------2) Create a method that takes in a hash and returns one array with all the hash values at their own index and in alphabetical order. No nested arrays. Use the test variable provided.
 # HINT: Google 'ruby get rid of nested arrays'
 
+
 us_states = { northwest: ['Washington', 'Oregon', 'Idaho'], southwest: ['California', 'Arizona', 'Nevada'], notheast: ['Maine', 'New Hampshire', 'Vermont'] }
 # Expected output: ['Arizona', 'California', 'Idaho', 'Maine', 'Nevada', 'New Hampshire', 'Oregon', 'Vermont', 'Washington'] 
 
+# flatten takes in multiple arrays and turns them into one array (ruby get rid of nested arrays)
+# use .sort to sort the states alphabetically
+
+p us_states.values.flatten.sort 
+
+# output: ["Arizona", "California", "Idaho", "Maine", "Nevada", "New Hampshire", "Oregon", "Vermont", "Washington"]
+
 
 # --------------------3a) Create a class called Bike that is initialized with a model, wheels, and current_speed. The default number of wheels is 2. The current_speed should start at 0. Create a bike_info method that returns a sentence with all the data from the bike object.
 
 # Expected output example: 'The Trek bike has 2 wheels and is going 0 mph.'
 
+class Bike 
+    attr_accessor :model, :wheels, :current_speed
+  
+    def initialize (model, wheels, current_speed)
+    @model = Trek
+    @wheels = 2
+    @current_speed = 0
+    end
+  
+    def bike_info
+       "The #{model} bike has #{wheels} and is going #{current_speed} mph."
```

Comment:
You will most likely need the @ symbol to access the instance variable in your string interpolation.